### DISCUSSION TITLE
Undocumented releases

### DISCUSSION THREAD

**Discussion by dariusc93** [libp2p/rust-libp2p#4774] [Q&A] [answered]
I have noticed that libp2p-swarm v0.43.7 and libp2p-request-response v0.25.3 were released a few days ago, but there was no indication in the changelog for these releases or anything from the commit history showing this change made to the respected `Cargo.toml` file. Could there be some explanation for these releases? While I could match the date of the release to the commit history, it doesn't really explain what may have changed that warranted the release.

**Comment by thomaseizinger** [comment]
Ah yes, good catch! I think we released those ones as backports.

See:

- https://github.com/libp2p/rust-libp2p/pull/4724
- https://github.com/libp2p/rust-libp2p/pull/4714

@mxinden How do we best get these into master but retain the commit SHA to have the tag as well? I don't think we have merge-commits enabled at all. Should we enable that and then do a regular merge of `v0.52` into `master`?

**Reply by dariusc93** [reply]
Thank you for the response

**Comment by mxinden** [comment, marked as answer]
Good catch @dariusc93. We should have done a better job on our release flow. In general, you can find a tag corresponding to each release, see https://github.com/libp2p/rust-libp2p/releases/tag/libp2p-request-response-v0.25.3. Note that these tags should as well be signed for authenticity.

https://github.com/libp2p/rust-libp2p/pull/4724 is already in `master` through https://github.com/libp2p/rust-libp2p/pull/4700. Ideally we would have merged https://github.com/libp2p/rust-libp2p/pull/4724 into `v0.52` first and then merged `v0.52` into `master`.

I don't think we need to get https://github.com/libp2p/rust-libp2p/pull/4714 into `master` given that `ConnectionHandlerEvent::Close` is removed with https://github.com/libp2p/rust-libp2p/pull/4755.

**Reply by dariusc93** [reply]
Thank you for the response
